### ISSUE TITLE
Issue with doxyfigcaption

### ISSUE THREAD

**Issue by zyphlar** [doxygen/doxygen#10707]
**Describe the bug**

When running `cd ./latex && make pdf` on a .dox file that includes images with captions, I get this error:

```
...
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/context/base/mkii\supp-pdf.mki
i
[Loading MPS to PDF converter (version 2006.09.02).]
)
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/epstopdf-pkg\epstopdf-ba
se.sty
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/00miktex\epstopdf-sys.cf
g))
*geometry* driver: auto-detecting
*geometry* detected driver: pdftex
No file refman.toc.
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/amsfonts\umsa.fd)
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/amsfonts\umsb.fd)
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/wasysym\uwasy.fd)
[1{C:/Users/me/AppData/Local/MiKTeX/fonts/map/pdftex/pdftex.map}{C:/Users/Wil
l/AppData/Local/Programs/MiKTeX/fonts/enc/dvips/base/8r.enc}] (intro.tex
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/psnfss\t1pcr.fd)
(C:\Users\me\AppData\Local\Programs\MiKTeX\tex/latex/psnfss\ts1phv.fd) <./figure.jpg> [1] <./image.jpg>
! Undefined control sequence.
\doxyfigcaption ->\H@refstepcounter
                                    {figure}\@dblarg {\@caption {figure}}
l.49 \doxyfigcaption
                    {hi}
?
```

My doxygen.sty is the autogenerated default:

```
% Define caption that is also suitable in a table
\makeatletter
\def\doxyfigcaption{%
\H@refstepcounter{figure}%
\@dblarg{\@caption{figure}}}
\makeatother
```

The offending .tex section:

```
\begin{DoxyImage}
\includegraphics[width=\textwidth]{image.jpg}
\doxyfigcaption{hi}
\end{DoxyImage}
```

And the offending .dox line:

```
\image latex image.jpg "hi" width=\textwidth
```

Changing the doxyfigcaption definition to empty resolves the error, just outputting the caption plainly without any kind of figure counters:

```
% Define caption that is also suitable in a table
\makeatletter
\def\doxyfigcaption{}
\makeatother
```

**Expected behavior**

Doxygen and LaTeX shouldn't explode on images with captions

**Version**
Latest from Github, Windows (1.11.0)

**Comment by albert-github** [comment]
I tried to reproduce the problem with the information provided, i.e. just the line:
```
\image latex image.jpg "hi" width=\textwidth
```
and a small `jpg` file that I had flying around but wit the current doxygen version (1.11.0 (42d0f98dcbb309dcae08d1495786add326aaeb41)) and current miktex version I was not able to reproduce the problem.

Also the by you mentioned version (1.11.0) is a floting version so the full version would be required (probably won't influence the issue though).

So:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by AndreasMatthias** [comment]
> ```
> ! Undefined control sequence.
> \doxyfigcaption ->\H@refstepcounter
>                                     {figure}\@dblarg {\@caption {figure}}
> ```

This command is defined in `hyperref.sty`. Do you see any warnings in the log file related to hyperref?

**Comment by zyphlar** [comment]
Ah I figured it out.

(a) `PDF_HYPERLINKS    = YES` must be set, this error happens with `PDF_HYPERLINKS    = NO` which might be a bug worth fixing.
(b) Changing the style to `\refstepcounter{figure}%` is also a valid workaround with seemingly no difference (is there supposed to be a link involved?)
(c) Previously my figure captions were like `Figure 1: Foo Bar` but now with this style/version/code/etc it's `Figure 1 Foo Bar` -- is there a good way to customize this or is this as intended?

**Comment by albert-github** [comment]
Thanks for the investigation, gave the right clue
- I've just pushed a proposed patch, pull request #10732

You wrote:
> Previously my figure captions were like ...

in which version was this and how should, in your opinion, the caption look like?
(probably it would be possible to define one's own caption style).

**Comment by AndreasMatthias** [comment]
> (c) Previously my figure captions were like `Figure 1: Foo Bar` but now with this style/version/code/etc it's `Figure 1 Foo Bar` -- is there a good way to customize this or is this as intended?

Doxygen uses `caption.sty`. The following should do the trick:
```
LATEX_EXTRA_STYLESHEET = my.sty
```
With file `my.sty`:
```latex
\AtBeginDocument{\captionsetup{labelsep=colon}}
```
See the documentation of `caption.sty` for further settings.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by zyphlar** [comment]
Thank you! Sorry I didn't see the messages earlier, this seems great.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
